### ISSUE TITLE
Error of mx.callback.do_checkpoint on Python3

### ISSUE THREAD

**Issue by Keiku** [dmlc/mxnet#2037]
When I executed mx.callback.do_checkpoint code on Python3, I got the below AttributeError because Python3 didn't support 'has_key'. Is there a way to use mx.callback.do_checkpoint on Python3?

```
Traceback (most recent call last):
  File "train_inception.py", line 86, in <module>
    epoch_end_callback = mx.callback.do_checkpoint(MODEL_PREFIX),
  File "/home/kuroyanagi/.pyenv/versions/anaconda3-2.3.0/lib/python3.4/site-packages/mxnet-0.5.0-py3.4.egg/mxnet/model.py", line 745, in fit
    self._init_params(dict(data.provide_data+data.provide_label))
  File "/home/kuroyanagi/.pyenv/versions/anaconda3-2.3.0/lib/python3.4/site-packages/mxnet-0.5.0-py3.4.egg/mxnet/model.py", line 501, in _init_params
    self.initializer(k, v)
  File "/home/kuroyanagi/.pyenv/versions/anaconda3-2.3.0/lib/python3.4/site-packages/mxnet-0.5.0-py3.4.egg/mxnet/initializer.py", line 111, in __call__
    if self.param.has_key(name):
AttributeError: 'dict' object has no attribute 'has_key'
```

**Comment by piiswrong** [comment]
if self.param.has_key(name) -> if name in self.param

**Comment by Keiku** [comment]
Thank you for your suggestion. I have created a PR for this issue just now.

**Comment by Keiku** [comment]
This issue was resolved by #2038.
